### ISSUE TITLE
Custom Colours: Fix broken functionality on Atomic sites.

### ISSUE THREAD

**Issue by kwight** [Automattic/wp-calypso#52759]
Custom Colours started as a feature on Simple sites, and was then quickly ported to `wpcomsh` for use by Atomic sites. It was a rush job (I know, I was part of it), and some of it might not be working properly after a lack of maintenance (or bugs missed in the original implementation).

Let's:
* Verify functionality on Simple (so that we know we have a working model to begin with).
* Compare Simple functionality to that on Atomic. Create issues for any broken pieces.

See also: p9F6qB-4rE-p2

<img width="400" alt="Screen Shot 2021-05-11 at 12 55 46 PM" src="https://user-images.githubusercontent.com/349751/117876468-4508c180-b258-11eb-8bc2-83a6952e090f.png">

Problems found on Seedlet:
* https://github.com/Automattic/wp-calypso/issues/54561
* https://github.com/Automattic/wp-calypso/issues/54563
* https://github.com/Automattic/wp-calypso/issues/54564

General problems:
* https://github.com/Automattic/wp-calypso/issues/54598

**Comment by kwight** [comment]
Related smaller task: 482-gh-Automattic/wpcomsh

**Comment by jordesign** [comment]
I"m happy to help with testing and verification between Simple/Atomic.

Do we want to approach this with certain theme/s as a baseline?

**Comment by kwight** [comment]
@jordesign Good question – Twenty Twenty One and Twenty Twenty come to mind, or maybe the Varia ("recommended") themes? @kathrynwp any suggestions for three themes to focus on during testing?

**Comment by kathrynwp** [comment]
In addition to Twenty Twenty One and Twenty Twenty, how about Hever, since it's in the signup flow and used by a ton of people? There is also currently a bug with previewing some of the colours in Hever even on Simple sites that would be great to fix at the same time. (for example, try to change the background colour in Hever - it's not visible in the preview area, only visible once the changes are saved).

**Comment by jordesign** [comment]
OK great - I'm EOD today - but tomorrow I can test:
* TwentyTwentyOne
* TwentyTwenty
* Seedlet
* Varia/Hever

**Comment by andres-blanco** [comment]
Twenty Twenty One and Twenty Twenty don't seem to exhibit this problem. The behaviour is the same on Atomic and Simple regarding the colors in the customizer. This might be due to the color section in the customizer being different in this themes, there's no "Color Palette" or color circles to pick:
![image](https://user-images.githubusercontent.com/375980/125662904-29bea033-4c49-4067-a88f-5a48c0f32b48.png)

I will keep searching for specific problems on the rest of themes.

**Comment by jordesign** [comment]
I'm going to skip testing Seedlet for now - based on those linked issues above.

**Comment by jordesign** [comment]
I started some testing for Hever - and it ended up being enough to warrant a P2...
pd2qbF-ck-p2

**Comment by DanielSmithMichigan** [comment]
I tried every recommended theme, and determined: Are custom colors completely/mostly broken for this theme on atomic?

My criteria was to change each color, and make sure it modified something on the page. I found that these were "completely/mostly broken:"

- Shawburn (Fixed by [4230](https://github.com/Automattic/themes/pull/4230))
- Morden (Fixed by [4230](https://github.com/Automattic/themes/pull/4230))
- Stow (Fixed by [4230](https://github.com/Automattic/themes/pull/4230))
- Hever (Fixed by [4230](https://github.com/Automattic/themes/pull/4230))
- Rockfield (Fixed by [4230](https://github.com/Automattic/themes/pull/4230))
- Seedlet (Fixed by [4236](https://github.com/Automattic/themes/pull/4236))
- Blank Canvas (Fixed by [4236](https://github.com/Automattic/themes/pull/4236))


Not completely broken but not really working either:
- Ryu: "Theme Color One" and "Theme Color Two" don't seem to do anything, but color schemes worked
- Hemingway Rewritten: "Main accent" didn't seem to do much, but color schemes worked
- Twenty Fifteen - Header and Sidebar Text color didn't seem to change anything

Other notes:
- For most themes, "link color" didn't change the color of the menu items. Is that expected behavior?
- Is it just me, or do most themes not really affect the look of my site that much?

**Comment by kwight** [comment]
> For most themes, "link color" didn't change the color of the menu items. Is that expected behavior?

No, that doesn't surprise me – that color is more for text links within content (often styled differently from the menu items in the header).

> Is it just me, or do most themes not really affect the look of my site that much?

Also not a surprise; themes support a varying degree of color customization, and subtle changes were not uncommon when we were writing all of the annotations.
